src/execution_plan/ops/op_update.c

```diff
@@ -85,10 +87,11 @@ OpResult OpUpdateConsume(OpBase *opBase, Record *r) {
         /* Find ref to property. */
         SIValue entry = Record_GetEntry(*r, update_expression->alias);
         GraphEntity *entity = (GraphEntity*) entry.ptrval;
+        AST_GraphEntity* ge = MatchClause_GetEntity(op->ast->matchNode, update_expression->alias);
```

Comment:
I suggest storing the type information somewhere within update_expressions,
that way we won't have to perform `MatchClause_GetEntity` every time consume is called.